parsec/core/gui/workspaces_widget.py

```diff
@@ -432,6 +433,13 @@ def unmount_workspace(self, workspace_id, timestamp=None):
             timestamp=timestamp,
         )
 
+    def update_workspace_config(self, workspace_id, state):
+        if state:
+            self.disabled_workspaces -= {workspace_id}
```

Comment:
.discard()/.add() ?

Reply:
We can't, `disabled_workspaces` is a `frozenset`.

Reply:
but `-=` is supposed to be an in-place operator... 

```
>>> x = {1, 2}
>>> y = x
>>> x -= {2}
>>> x
{1}
>>> y
{1}
>>> x = frozenset({1})
{}>>> x
frozenset({1})
>>> y = x
>>> x += {2}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +=: 'frozenset' and 'set'
>>> x -= {2}
>>> x
frozenset({1})
>>> y
frozenset({1})
>>> x -= {1}
>>> x
frozenset()
>>> y
frozenset({1})
```

O_o

This feels really inconsistent to me, I would prefer to use `disabled_workspaces.difference({workspace_id})` and pretend I've never witnessed this whole craziness ^^

Reply:
You can't add sets with `+`, you're supposed to `or` them
```python
>>> {1} + {2}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +=: 'set' and 'set'
>>> {1} | {2}
{1, 2}
```
> but -= is supposed to be an in-place operator...

But "in-place" doesn't mean mutating the existing data structure:
```python
>>> a = (1,)                                                                              
>>> a += (2,)                                                                             
>>> a                                                                                     
(1, 2)

>>> a = 1                                                                                 
>>> a += 2                                                                                
>>> a                                                                                     
3
```